core/state/journal.go

```diff
@@ -174,6 +177,17 @@ func (ch resetObjectChange) dirtied() *common.Address {
 	return ch.account
 }
 
+func (ch sendallChange) revert(s *StateDB) {
+	obj := s.getStateObject(*ch.account)
+	if obj != nil {
+		obj.sendalled = false
+	}
+}
+
+func (ch sendallChange) dirtied() *common.Address {
+	return nil // TODO: return nil or account here?
```

Comment:
This must return `ch.account`. The `dirtied` is used later, during statedb Finalize, to know what accounts that have been modified during execution. So if the `sendall` change _is_ a state-modifying operation, then it should return the `ch.account`. However, if we consider the balance-change as a separate event (that is, a separate journalled event), then the `sendallChange` is not state-modifying.